source/create.ts

```diff
@@ -26,21 +61,23 @@ const create = defaults => {
 		defaults.handler = (options, next) => next(options);
 	}
 
-	function got(url, options?: any) {
+	// @ts-ignore Because the for loop handles it for us, as well as the other Object.defines
+	const got: Got = (url: URL | string | Partial<Options>, options?: Partial<Options>): ProxyStream | CancelableRequest<Response> => {
```

Comment:
Why are you using `Partial` for every `Options` usage?

Reply:
Partial is a type that marks every property of a type as optional (so undefined is valid)

Then again, most things in the Options interface ARE optional, so they could be useless..

Reply:
I know how `Partial` works, I was more curious why it's used everywhere when the `Options` properties are already optional.

Reply:
Not *everything* in the Options object is optional, which is one of the reasons I made it use Partial. I could make everything in options BE optional, if that'd be correct

Reply:
> Not everything in the Options object is optional

IMO everything should be optional there. Only `NormalizedOptions` requires all options to be defined.

Reply:
Everything in `Options` should be optional.